articles/140_resource_names.md

```diff
@@ -192,81 +217,153 @@ Substitutions are also not allowed to be nested, i.e. substitutions may not cont
+#### Hidden nodes
 
-## Mapping of ROS 2 Topic and Service Names to DDS Concepts
+If a (resource provider) node name contains any tokens that start with an underscore (`_`), the node it refers to will be considered hidden and tools may not show them unless explicitly asked to.
 
-The ROS topic and service name constraints allow more types of characters than the DDS topic names because ROS additionally allows the forward slash (`/`), the tilde (`~`), and the balanced curly braces (`{}`).
-These must be substituted or otherwise removed during the process of converting the topic or service name to DDS concepts.
-Since ROS 2 topic and service names are expanded to fully qualified names, any balanced bracket (`{}`) substitutions and tildes (`~`) will have been expanded.
-Additionally any URL related syntax, e.g. the `rostopic://` prefix, will be removed once parsed.
-Previously forward slashes (`/`) were disallowed in DDS topic names, now the restriction has been lifted (see [issue](https://issues.omg.org/issues/lists/dds-rtf5#issue-42236) on omg.org) and therefore the ROS topic names are first prefixed with ROS Specific Namespace prefix (described below) are then mapped directly into DDS topic names.
+#### Hidden resources
+
+If a resource name contains any tokens that start with an underscore (`_`), the resource it refers to will be considered hidden and tools may not show them unless explicitly asked to.
+
+### Examples
+
+For example, these are valid names:
+
+| `foo`      | `abc123`   | `_foo`  | `Foo`               | `BAR`                |
+| `~`        | `foo/bar`  | `~/foo` | `{foo}_bar`         | `foo/{ping}/bar`     |
+| `foo/_bar` | `foo_/bar` | `foo_`  | `rosservice:///foo` | `rostopic://foo/bar` |
+| `foo/_bar` | `foo_/bar` | `foo_`  | `rosservice:///foo` | `rostopic://foo/bar` |
+
+But these are not valid names:
+
+| `123abc`    | `123`  | `foo bar`  | ` `        | `foo//bar` |
+| `/~`        | `~foo` | `foo~`     | `foo~/bar` | `foo/~bar` |
+| `foo/~/bar` | `foo/` | `foo__bar` |            |            |
+
+These are some valid fully qualified names:
+
+| `/foo`                    | `rosparam://another.node/bool_param` | `rostopic:///ping`                 | `/_private/thing`                  |
+| `rosaction://a.node/ping` | `/bar/baz`                           | `/public_namespace/_private/thing` | `rosservice://_private.node/reset` |
+
+### Summary
+
+For convenience, here is a summary of all the rules that apply to URLs in ROS.
+
+Provider names in a URL:
+
+- may contain alphanumeric characters (`[0-9|a-z|A-Z]`), underscores (`_`) and periods (`.`).
+- must not start with a numeric character (`[0-9]`).
+- must not end nor start with a period (`.`).
+- must not contain any number of repeated periods (`.`).
+- must not contain any number of repeated underscores (`_`).
```

Comment:
Made this constraint explicit everywhere else in 7680627ee8f2fa8675fd58e72f814585c6721f2a. Do you think we shouldn't?

Reply:
Hmm, not sure. I think we only _needed_ it for the alternative we're no longer using. I don't think the code enforces it at the moment. Maybe it should be removed. @sloretz or anyone on the @ros2/team do you have an opinion here?

Reply:
I don't have a strong preference. But it seems okay to remove the constraint if there's no reason to have it. If we really need it in the future, we could start using repeated `.` as a delimiter since it's already disallowed.